discourse/discourse.dockerfile

```diff
@@ -0,0 +1,35 @@
+# Based on Discourse's https://github.com/discourse/discourse/blob/master/docs/DEVELOPER-ADVANCED.md
+# Does not use RVM because we're just running Discourse on here.
+FROM janx/ubuntu-dev
+MAINTAINER Jan Keromnes "janx@linux.com"
+MAINTAINER Michael Howell "michael@notriddle.com"
+
+# Download deps
+RUN sudo add-apt-repository ppa:chris-lea/redis-server && \
+    sudo add-apt-repository ppa:brightbox/ruby-ng && \
+    curl -sL https://deb.nodesource.com/setup_8.x | sudo -E bash - && \
+    sudo apt-get -yqq update && \
+    sudo apt-get -yqq install nodejs ruby2.3 python-software-properties vim curl expect debconf-utils git-core build-essential zlib1g-dev libssl-dev openssl libcurl4-openssl-dev libreadline6-dev libpcre3 libpcre3-dev imagemagick postgresql postgresql-contrib-9.5 libpq-dev postgresql-server-dev-9.5 redis-server advancecomp gifsicle jhead jpegoptim libjpeg-turbo-progs optipng pngcrush pngquant gnupg2 ruby && \
+    echo 'gem: --no-document' >> /home/user/.gemrc && \
+    sudo gem install bundler mailcatcher && \
+    sudo -u postgres createuser --superuser -Upostgres user && \
+    sudo -u postgres psql -c "ALTER USER user WITH PASSWORD 'password';" && \
+    sudo npm install -g svgo phantomjs-prebuilt
+
+# Download and configure Discourse
+RUN git clone https://github.com/discourse/discourse && \
```

Comment:
Nit: It's slightly ambiguous where the current directory is, so please use absolute paths, e.g. `git clone https://github.com/discourse/discourse /home/user/discourse`.